### ISSUE TITLE
Concurrency chapter - example of goroutine loop variables is no longer applicable as of go version 1.22 

### ISSUE THREAD

**Issue by rqpt** [quii/learn-go-with-tests#740]
tldr - This behaviour is something that has been addressed in the latest go release.

I am currently reading this book (excellent btw), and am coding along until I get to the '...and we're back' section of the concurrency chapter.

![tdd-go-concurrency-chapter](https://github.com/quii/learn-go-with-tests/assets/76003847/432aee20-a07b-4e15-b896-562335a796b5)

I ran this example and my tests pass just fine. 

![tdd-go-concurrency-editor](https://github.com/quii/learn-go-with-tests/assets/76003847/bb959290-fa43-405c-84ef-2fad8ddb3a34)

I then set out to investigate why, as this should fail according to the chapter, and this video cleared things up for me - 
https://www.youtube.com/watch?v=Smc-6qfZ2JU

**Comment by quii** [comment]
Good spot! Do you fancy making a PR?

**Comment by rqpt** [comment]
I'll give it a shot, sure! I'll try and tackle it this weekend.

**Comment by primeapple** [comment]
I took some minutes and added a PR for this.
